src/query/v2/plan/operator.cpp

```diff
@@ -2376,6 +2376,22 @@ class DistributedCreateExpandCursor : public Cursor {
     return true;
   }
 
+  void PullMultiple(MultiFrame &multi_frame, ExecutionContext &context) override {
```

Comment:
How does this work without sending bool?

Reply:
The bool that is returned previously can be "got" from the multi_frame itself for the single cases: if it has any valid frames, then it the bool is true, otherwise it is false.

The not so okay case if we push down a multi-frame that already has data (e.g.: `OPTIONAL MATCH` will get a multi-frame from its left side branch, then push down the same multiframe to the right side and we expect it to conserve the information. However the solution heavily depends on how we implement the optional operator, thus I wouldn't like to focus on how this logic should work. Our primary goal now is to make vertex/edge creation and retrieval work and see how does it perform. After that comes the aggregation and optional match probably.